synapse/rest/admin/rooms.py

```diff
@@ -237,3 +238,80 @@ def __init__(self, hs):
                 response["prev_batch"] = 0
 
         return 200, response
+
+
+class JoinRoomAliasServlet(RestServlet):
+
+    PATTERNS = admin_patterns("/join/(?P<room_identifier>[^/]*)")
+
+    def __init__(self, hs):
+        self.hs = hs
+        self.auth = hs.get_auth()
+        self.room_member_handler = hs.get_room_member_handler()
+        self.admin_handler = hs.get_handlers().admin_handler
+        self.state_handler = hs.get_state_handler()
+
+    async def on_POST(self, request, room_identifier):
+        requester = await self.auth.get_user_by_req(request)
+        await assert_user_is_admin(self.auth, requester.user)
+
+        try:
+            content = parse_json_object_from_request(request)
+        except Exception:
+            # Turns out we used to ignore the body entirely, and some clients
+            # cheekily send invalid bodies.
+            content = {}
```

Comment:
This looks like a copy-paste from another servlet. If `content` is `{}` here, the user is going to get an exception anyways (that they didn't provide a `user_id` key, which doesn't seem like it would be much more useful than that raised by `parse_json_object_from_request`.

I'd probably return an `M_BAD_JSON` exception here, with the error.